File: go/internal/app/bitmex/models/settlement.go
package models

import "time"

// Settlement Historical Settlement Data
type Settlement struct {
	Bankrupt              int64     `json:"bankrupt,omitempty"`
	OptionStrikePrice     float64   `json:"optionStrikePrice,omitempty"`
	OptionUnderlyingPrice float64   `json:"optionUnderlyingPrice,omitempty"`
	SettledPrice          float64   `json:"settledPrice,omitempty"`
	SettlementType        string    `json:"settlementType,omitempty"`
	Symbol                string    `json:"symbol"`
	TaxBase               int64     `json:"taxBase,omitempty"`
	TaxRate               float64   `json:"taxRate,omitempty"`
	Timestamp             time.Time `json:"timestamp"`
}

File: ts/src/internal/app/bitmex/models/settlement.ts
// Settlement Historical Settlement Data
export default interface Settlement {
  bankrupt: number;
  optionStrikePrice: number;
  optionUnderlyingPrice: number;
  settledPrice: number;
  settlementType: string;
  symbol: string;
  taxBase: number;
  taxRate: number;
  timestamp: Date;
}
